x-pack/plugins/observability_solution/inventory/public/pages/inventory_page/index.tsx

```diff
@@ -4,105 +4,74 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EuiDataGridSorting } from '@elastic/eui';
-import React from 'react';
-import useEffectOnce from 'react-use/lib/useEffectOnce';
-import { EntityColumnIds, EntityType } from '../../../common/entities';
-import { EntitiesGrid } from '../../components/entities_grid';
-import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
-import { useInventoryAbortableAsync } from '../../hooks/use_inventory_abortable_async';
+import React, { useCallback, useEffect, useState } from 'react';
+import { decode, encode } from '@kbn/rison';
+import { ENTITY_TYPE } from '@kbn/observability-shared-plugin/common';
+import { GroupedInventoryView } from '../../components/grouped_inventory';
+import { UngroupedInventoryView } from '../../components/grouped_inventory/ungrouped_view';
 import { useInventoryParams } from '../../hooks/use_inventory_params';
 import { useInventoryRouter } from '../../hooks/use_inventory_router';
-import { useKibana } from '../../hooks/use_kibana';
+import { InventoryPageViewContextProvider } from '../../context/inventory_page_view_provider';
+
+const DEFAULT_GROUPING = ENTITY_TYPE;
+
+export interface InventoryState {
+  groupBy?: string;
+  pagination?: Record<string, number>;
+}
 
 export function InventoryPage() {
-  const { searchBarContentSubject$ } = useInventorySearchBarContext();
-  const {
-    services: { inventoryAPIClient },
-  } = useKibana();
   const { query } = useInventoryParams('/');
-  const { sortDirection, sortField, pageIndex, kuery, entityTypes } = query;
-
   const inventoryRoute = useInventoryRouter();
 
-  const {
-    value = { entities: [] },
-    loading,
-    refresh,
-  } = useInventoryAbortableAsync(
-    ({ signal }) => {
-      return inventoryAPIClient.fetch('GET /internal/inventory/entities', {
-        params: {
-          query: {
-            sortDirection,
-            sortField,
-            entityTypes: entityTypes?.length ? JSON.stringify(entityTypes) : undefined,
-            kuery,
-          },
-        },
-        signal,
-      });
-    },
-    [entityTypes, inventoryAPIClient, kuery, sortDirection, sortField]
+  const { view } = query;
```

Comment:
`view` is a string now, you can create an io-ts codec to automatically transform it to `InventoryState` and define it in the route definition. You can use the [entityTypesRt](https://github.com/Bluefinger/kibana/blob/2403e4d4cadd025316d80b9a7a0faf78a44596e3/x-pack/plugins/observability_solution/inventory/public/routes/config.tsx#L32) as example. You won't need to manually call `decode`.

Reply:
Unfortunately, I keep running into problems with io-ts and trying to express a partial record `Partial<Record<EntityView, number>>` just causes me all sorts of grief. Perhaps I use Zod here instead?

Reply:
>Perhaps I use Zod here instead?

I don't think we should mix the two RT types.